### PR TITLE
Add boolector

### PR DESCRIPTION
**Pull request by tmeissner**
Attempt to add Boolector.

### CODE REVIEW

File: boolector.dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    binutils \
+    cmake \
+    autoconf \
+    git \
+ && apt-get autoclean && apt-get clean && apt-get -y autoremove \
+ && update-ca-certificates  \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir /tmp/boolector && cd /tmp/boolector \
+&& curl -fsSL https://codeload.github.com/boolector/boolector/tar.gz/master | tar xzf - --strip-components=1 \
```
Comment (eine) [inline boolector.dockerfile:14]: Please add one space at the beginning of L14-L24.

File: boolector.dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+&& ./contrib/setup-btor2tools.sh \
+&& ./contrib/setup-lingeling.sh \
+&& ./configure.sh \
+&& make -C build -j$(nproc) PREFIX=/opt/boolector \
+&& mkdir -p /opt/boolector/usr/local/bin \
```
Comment (eine) [inline boolector.dockerfile:19]: Maybe?

```
&& mkdir -p /opt/boolector/usr/local/bin /opt/boolector/usr/local/lib \
&& cp build/bin/boolector build/bin/btor* deps/btor2tools/bin/btorsim /opt/boolector/usr/local/bin/ \
&& cp build/lib/libboolector.a /opt/boolector/usr/local/lib/
```

**Review by eine — COMMENTED** [review]
Looks good!

Do you want to try adding it to the graph? See yices2: https://github.com/hdl/containers/commit/dc677af6914af9a77ae5891f14a595c07b0df9e5

**Comment by eine** [comment]
Please, remove boolector from the To Do list in `doc/index.adoc`!

File: boolector.dockerfile (hunk 1 above)
Comment (tmeissner) [reply boolector.dockerfile:14]: Sure.

> Looks good!
> 
> Do you want to try adding it to the graph? See yices2: [dc677af](https://github.com/hdl/containers/commit/dc677af6914af9a77ae5891f14a595c07b0df9e5)

I will try.

File: boolector.dockerfile (hunk 2 above)
Comment (tmeissner) [reply boolector.dockerfile:19]: Yep, looks better now.

**Comment by tmeissner** [comment]
> Please, remove boolector from the To Do list in `doc/index.adoc`!

Done.

File: boolector.dockerfile (hunk 1 above)
Comment (tmeissner) [reply boolector.dockerfile]: I tried to add `boolector` to the graph. Furthermore I fixed the `yices2` & `z3` `base` images to `:build`. Maybe you can have a look.

**Comment by eine** [comment]
Awesome! Thanks!

**Pull request merged**